scanner/scanner.go

```diff
@@ -777,6 +777,15 @@ func (s *Scanner) scanComment(ctx *Context) bool {
 func (s *Scanner) scanMultiLine(ctx *Context, c rune) error {
 	state := ctx.getMultiLineState()
 	ctx.addOriginBuf(c)
+	// normalize CR and CRLF to LF
+	if c == '\r' {
+		if ctx.nextChar() == '\n' {
+			ctx.addOriginBuf('\n')
+			s.progress(ctx, 1)
+			s.offset++
+		}
+		c = '\n'
+	}
```

Comment:
I'm not fully confident about this change, since most of the logic is quite similar to [what's in the scanNewLine method](https://github.com/goccy/go-yaml/blob/ac837bc4af85af2051db79efb81d5d41e924219e/scanner/scanner.go#L874-L879). I tried to refactor and deduplicate it, but couldn’t come up with a better approach. Open to any suggestions 🙇